### PR TITLE
deepin.dde-polkit-agent: set plugins dir from environment variable

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

deepin.dde-polkit-agent: set plugins dir from environment variable

About packaging deepin for NixOS: https://github.com/NixOS/nixpkgs/issues/59023

See also https://github.com/NixOS/nixpkgs/pull/59244#issuecomment-485097016

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/deepin/dde-polkit-agent/dde-polkit-agent.plugins-dir.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+diff --git a/pluginmanager.cpp b/pluginmanager.cpp
+index 0c03237..9c46eb9 100644
+--- a/pluginmanager.cpp
++++ b/pluginmanager.cpp
+@@ -34,7 +34,8 @@ QList<QButtonGroup*> PluginManager::reduceGetOptions(const QString &actionID)
+ void PluginManager::load()
+ {
+ 
+-    QDir dir("/usr/lib/polkit-1-dde/plugins/");
++    QDir dir(QProcessEnvironment::systemEnvironment().value("DDE_POLKIT_PLUGINS_DIR", "/usr/lib/polkit-1-dde/plugins/"));
```
Comment (worldofpeace) [inline pkgs/desktops/deepin/dde-polkit-agent/dde-polkit-agent.plugins-dir.patch]: Can you give me an example or a source where DDE ships a plugin like this for polkit?

Comment (romildo) [reply pkgs/desktops/deepin/dde-polkit-agent/dde-polkit-agent.plugins-dir.patch]: > Can you give me an example or a source where DDE ships a plugin like this for polkit?

`lib/polkit-1-dde/plugins/libdpa-ext-gnomekeyring.so` from [`dpa-ext-gnomekeyring`](https://github.com/linuxdeepin/dpa-ext-gnomekeyring)

**Comment by romildo** [comment]
Modified the patch so that the environment variable may be a list of directories to look for plugins. This is much more flexible and will facilitate wrapping.

**Comment by romildo** [comment]
Submitted upstream: https://github.com/linuxdeepin/dde-polkit-agent/pull/12

**Pull request merged**